### PR TITLE
Update WASM binary and Land WASM rust sources in this repository

### PR DESCRIPTION
**Pull request by ochameau**
Today, the rust sources used to [generate lib/mappings.wasm](https://github.com/mozilla/source-map/blob/master/lib/mappings.wasm) are hosted on:
https://github.com/fitzgen/source-map-mappings/
This looks unofficial, whereas that's where the sources come from and last binary was built from that repo.

It would be easier to fold the rust sources in this repo to ease contributions to it, while increasing visibility to the rust sources.

While we are at landing sources in-tree, it would also be nice to refresh the wasm binary by building it with latest rust version.
The current binary has been being in 2018 and is no longer reproducible.

In an effort to help reproduce the same wasm file, here is the environment from which I built it:
```
$ cargo -V
cargo 1.64.0 (387270bc7 2022-09-16)
$ rustup -V
rustup 1.25.1 (bb60b1e89 2022-07-12)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.64.0 (a55dd71d5 2022-09-19)`
```
Using instruction documented [here](https://github.com/mozilla/source-map/blob/master/CONTRIBUTING.md#updating-the-libmappingswasm-webassembly-module).

### CODE REVIEW

File: wasm-mappings/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+README.md -diff -merge
```
[the remark this answers is not in the document]
Comment (ochameau) [reply wasm-mappings/.gitattributes]: Oh yes I forgot to remove it after folding it into the main one.

File: wasm-mappings/LICENSE-APACHE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                              Apache License
```
[the remark this answers is not in the document]
Comment (ochameau) [reply wasm-mappings/LICENSE-APACHE:1]: I'll reach out before landing. I suspect they changed this repo license without changing this other repo.

**Comment by ochameau** [comment]
> RSLGTM on the Rust and Python code (assuming that it's just copying existing work).

Yes I copied master branch as is.

File: wasm-mappings/LICENSE-APACHE (hunk 2 above)
[the remark this answers is not in the document]
Comment (fitzgen) [reply wasm-mappings/LICENSE-APACHE:1]: I am happy to relicense the rust code as BSD.

[the remark this answers is not in the document]
Comment (ochameau) [reply wasm-mappings/LICENSE-APACHE:1]: Thanks a lot for chiming in Nick!

**Comment by ochameau** [comment]
I removed the license files from wasm folder, updated various files to mention BSD license, and updated github repo reference to mozilla's source-map one.

File: wasm-mappings/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Contributing to `source-map-mappings`
+
+Hi! We'd love to have your contributions! If you want help or mentorship, reach
+out to us in a GitHub issue, or ping `fitzgen` in [#rust on irc.mozilla.org](irc://irc.mozilla.org#rust)
```
Comment (bomsy) [inline wasm-mappings/CONTRIBUTING.md:4]: Is this irc channel still valid?

File: wasm-mappings/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# `source-map-mappings`
```
Comment (bomsy) [inline wasm-mappings/README.md:1]: Sould this become `wasm-mappings`?

**Review by bomsy — APPROVED** [review]
LGTM! Just had comments/questions about documentation updates
Thanks

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -152,11 +152,11 @@ $ rustup toolchain install nightly
 $ rustup target add wasm32-unknown-unknown --toolchain nightly
 ```
 
-Next, clone the Rust source used to create `lib/mappings.wasm`:
+Move to wasm sources folder:
 
 ```
 $ git clone https://github.com/fitzgen/source-map-mappings.git
```
[the remark this answers is not in the document]
Comment (ochameau) [reply CONTRIBUTING.md]: yes, forgot to remove the command line!

File: wasm-mappings/CONTRIBUTING.md (hunk 3 above)
Comment (ochameau) [reply wasm-mappings/CONTRIBUTING.md]: irc server is down, I removed it with the new matrix channel.

File: wasm-mappings/README.md (hunk 4 above)
Comment (ochameau) [reply wasm-mappings/README.md:1]: No, the rust package name is still source-map-mappings (you can see it referenced in cargo files)

**Comment by ochameau** [comment]
I've also updated the patch to ensure using latest master revision of source-map-mappings.
So that we are using https://github.com/fitzgen/source-map-mappings/commit/197660faf3dd1167c643797a5ed62cb687482c54 here, with the few cleanups and renames.

**Comment by ochameau** [comment]
With the latest changeset refreshing the wasm binary with latest version of rust, there seem to be a significant performance improvement:
https://treeherder.mozilla.org/perfherder/comparesubtest?originalProject=try&newProject=try&newRevision=340b70fc0526e767e8657b8f41749240d14d2908&originalSignature=4081483&newSignature=4081483&framework=12&originalRevision=b4fda98f266ca9b53c8f3962d6e92a9d68e32b57&page=1&showOnlyImportant=1
image
(I ported the existing perf test from bench folder to firefox CI)

**Review by eemeli — APPROVED** [review]
Looks good, provided that the contribution guidelines and the README link to them is fixed.

**Comment by ochameau** [comment]
When testing against Babel and React sourcemaps there is also significant improvements when running on Firefox:
image
Up to 20% faster when parsing the babel sourcemap and between 5% to 7% faster on mapping generated to original locations.

File: wasm-mappings/README.tpl [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# `{{crate}}`
+
+{{readme}}
```
[the remark this answers is not in the document]
Comment (ochameau) [reply wasm-mappings/README.tpl:3]: I imagine there must be benefits to use this setup when more than one file contribute to the README.md.
But in the current state of things, I agree it sounds easier.

File: wasm-mappings/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
+# Contributing to `source-map-mappings`
```
[the remark this answers is not in the document]
Comment (ochameau) [reply wasm-mappings/CONTRIBUTING.md:1]: I dropped the redundant content, but kept it for wasm specifics instructions.
I may consider moving "updating mappings.wasm" here?
I find it nice to keep the rust/wasm part of the codebase still contained in this subfolder.

[the remark this answers is not in the document]
Comment (eemeli) [reply wasm-mappings/CONTRIBUTING.md:1]: Yeah, that makes sense.

**Comment by ochameau** [comment]
I had to request the enabling of `actions-rs/toolchain@v1` on mozilla repos for the wasm github actions:
https://bugzilla.mozilla.org/show_bug.cgi?id=1797457
Otherwise the github actions for wasm were failing on rust tests.

Now everything is green and good to go!

**Pull request merged**